app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -155,13 +161,21 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
         super.onStart()
 
         sessionFeature?.start()
+        serviceLocator!!.sessionRepo.events.subscribe {
+            when (it) {
+                SessionRepo.Event.YouTubeBack -> youtubeBackHandler.handleBackClick()
+                SessionRepo.Event.ExitYouTube -> youtubeBackHandler.goBackBeforeYouTube()
+                null -> return@subscribe
+            }.forceExhaustive
+        }.addTo(compositeDisposable)
     }
 
     override fun onStop() {
         super.onStop()
 
         serviceLocator!!.sessionRepo.exitFullScreenIfPossible()
         sessionFeature?.stop()
+        compositeDisposable.clear()
```

Comment:
Is it better to clear the compositeDisposable in bulk (like here in `onStop`) vs as we receive each event?

Reply:
`Disposable` is equivelent to a coroutine `Job`, so it doesn't hold references to passed events.  Instead, `compositeDisposable.clear()` is basically `List<Job>.forEach { it.cancel() }`